libflux/flux-core/src/semantic/env.rs

```diff
@@ -141,6 +141,17 @@ impl Environment {
         self.lookup(&Symbol::from(v))
     }
 
+    pub(crate) fn lookup_symbol(&self, v: &str) -> Option<&Symbol> {
+        // TODO Avoid iteration here
```

Comment:
Iteration is expensive compared to using the actual maps lookup. Just can't do it with strings directly. Probably need to refactor this but I can't be bothered until I get the rest working.

Reply:
Oh, I guess this may be a big confusing as `Symbol` is just a `Rc<str>` in this PR, Before and after this the commits submitted here I have either a local id (`u32`) or the package name in `Symbol` which is why looking up via `&str` and `Symbol` is expected to differ.

Reply:
I'd like to split `Environment` into a type used internally during inference etc and another map-like type which is used externally and represents the exported APi (and therefore goes by string lookup). Currently `Environment` is used in both which makes it a bit awkward.